proxy/src/telemetry/sensor/http.rs

```diff
@@ -203,17 +222,25 @@ where
     }
 
     fn call(&mut self, mut req: Self::Request) -> Self::Future {
-        let (inner, body_inner) = match req.extensions_mut().remove::<Arc<ctx::transport::Server>>() {
-            None => (None, None),
-            Some(ctx) => {
+        let metadata = (
+            req.extensions_mut().remove::<Arc<ctx::transport::Server>>(),
+            req.extensions_mut().remove::<RequestOpen>()
+        );
+        let (inner, body_inner) = match metadata {
+            (ctx @ None, request_open) | (ctx, request_open @ None) => {
```

Comment:
alternatively, just match on the `(Some(), Some())` case first and fall back to `(ctx, request_open)`

Reply:
Yeah, that's much clearer, changed in 0b6014fcaae38aa8f2e53626f75ac0c72d20d458